src/ApacheOrcDotNet.OptimizedReader/Buffers/BaseColumnBuffer.cs

```diff
@@ -101,7 +81,90 @@ private protected void ReadBooleanStream(StreamDetail stream, ReadOnlySpan<byte>
                 return;
 
             var numSkipped = 0;
-            var bufferReader = new BufferReader(ResizeBuffer(stream, buffer, length));
+            var bufferReader = new BufferReader(GetDataStream(stream, buffer, length));
+            var numOfTotalBitsToSkip = stream.Positions.ValuesToSkip * 8 + stream.Positions.RemainingBits;
+            var numOfBytesToSkip = numOfTotalBitsToSkip / 8;
+            while (!bufferReader.Complete)
+            {
+                var numByteValuesRead = OptimizedByteRLE.ReadValues(ref bufferReader, _boolStreamBuffer);
+
+                for (int idx = 0; idx < numByteValuesRead; idx++)
+                {
+                    if (numSkipped++ < numOfBytesToSkip)
+                        continue;
+
+                    var decodedByte = _boolStreamBuffer[idx];
+                    var isFinalByte = bufferReader.Complete && idx >= numByteValuesRead - 1;
+
+                    // Skip remaining bits.
+                    if (numOfBytesToSkip % 8 != 0)
```

Comment:
This check was removed as well but an extra set of checks needed to be included to skip values from the remaining bits as well. This change was made because the bit shifting logic could erroneously inject the amount of remaining bits as `false` values in the current byte.